ui/component/livestreamChatLayout/view.jsx

```diff
@@ -0,0 +1,375 @@
+
+import { formatLbryUrlForWeb } from 'util/url';
+import { Menu, MenuButton, MenuList, MenuItem } from '@reach/menu-button';
+import { useHistory } from 'react-router-dom';
+import { useIsMobile } from 'effects/use-screensize';
+import * as ICONS from 'constants/icons';
+import Button from 'component/button';
+import classnames from 'classnames';
+import CommentCreate from 'component/commentCreate';
+import CreditAmount from 'component/common/credit-amount';
+import Icon from 'component/common/icon';
+import LivestreamComment from 'component/livestreamComment';
+import LivestreamComments from 'component/livestreamComments';
+import LivestreamSuperchats from 'component/livestreamSuperchats';
+import React from 'react';
+import Spinner from 'component/spinner';
+import Yrbl from 'component/yrbl';
+
+const IS_TIMESTAMP_VISIBLE = () =>
+  // $FlowFixMe
+  document.documentElement.style.getPropertyValue('--live-timestamp-opacity') === '0.5';
+
+const TOGGLE_TIMESTAMP_OPACITY = () =>
+  // $FlowFixMe
+  document.documentElement.style.setProperty('--live-timestamp-opacity', IS_TIMESTAMP_VISIBLE() ? '0' : '0.5');
+
+const VIEW_MODES = {
+  CHAT: 'chat',
+  SUPERCHAT: 'sc',
+};
+const COMMENT_SCROLL_TIMEOUT = 25;
+const LARGE_SUPER_CHAT_LIST_THRESHOLD = 20;
+
+type Props = {
+  claim: ?StreamClaim,
+  comments: Array<Comment>,
+  embed?: boolean,
+  isPopoutWindow?: boolean,
+  pinnedComments: Array<Comment>,
+  superChats: Array<Comment>,
+  uri: string,
+  doCommentList: () => void,
+  doCommentSocketConnect: (string) => void,
+  doCommentSocketDisconnect: (string) => void,
+  doResolveUris: (Array<string>) => void,
+  doSuperChatList: () => void,
+};
+
+export default function LivestreamChatLayout(props: Props) {
+  const {
+    claim,
+    comments: commentsByChronologicalOrder,
+    embed,
+    isPopoutWindow,
+    pinnedComments,
+    superChats: superChatsByAmount,
+    uri,
+    doCommentList,
+    doCommentSocketConnect,
+    doCommentSocketDisconnect,
+    doResolveUris,
+    doSuperChatList,
+  } = props;
+
+  const {
+    location: { pathname },
+  } = useHistory();
+
+  const isMobile = useIsMobile();
+
+  const discussionElement = document.querySelector('.livestream__comments');
+
+  const restoreScrollPos = React.useCallback(() => {
+    if (discussionElement) discussionElement.scrollTop = 0;
+  }, [discussionElement]);
+
+  const commentsRef = React.createRef();
+
+  const [viewMode, setViewMode] = React.useState(VIEW_MODES.CHAT);
+  const [scrollPos, setScrollPos] = React.useState(0);
+  const [showPinned, setShowPinned] = React.useState(true);
+  const [resolvingSuperChats, setResolvingSuperChats] = React.useState(false);
+  const [mention, setMention] = React.useState();
+  const [openedPopoutWindow, setPopoutWindow] = React.useState(false);
+  const [chatHidden, setChatHidden] = React.useState(false);
+
+  const quickMention =
+    mention && formatLbryUrlForWeb(mention).substring(1, formatLbryUrlForWeb(mention).indexOf(':') + 3);
```

Comment:
Channel mentions used to be canonical, but now it seems to be fixed at 2 characters?
![image](https://user-images.githubusercontent.com/64950861/149170148-dd7f04ae-8f81-417f-a5d3-ca35144fccc5.png)

Reply:
This is from the new inline mention feature. It should resolve + get canonical, Rafael is aware.